src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -236,12 +236,32 @@ private static bool PointerCastDefinitelyCantBeRemoved(CastExpressionSyntax cast
             return false;
         }
 
-        private static bool EnumCastDefinitelyCantBeRemoved(CastExpressionSyntax cast, ITypeSymbol expressionType)
+        private static bool EnumCastDefinitelyCantBeRemoved(CastExpressionSyntax cast, ITypeSymbol expressionType, ITypeSymbol castType)
```

Comment:
would love to rename this to 'Cannot' :)

Reply:
also, would prefer EnumCastCouldBeRemoved, and the caller checks the negation and bails.

Reply:
> would love to rename this to 'Cannot'

The current pattern is consistent with other methods in the file. (Outside the scope of this pull request.)

> also, would prefer EnumCastCouldBeRemoved, and the caller checks the negation and bails.

Both patterns are present. It may be possible to clean this up, but it's outside the scope of this pull request.